primitives/blockchain/src/error.rs

```diff
@@ -17,142 +17,143 @@
-	#[display(fmt = "Remote node has responded with invalid header proof")]
+	#[error("Remote node has responded with invalid header proof")]
 	InvalidCHTProof,
 	/// Remote fetch has been cancelled.
-	#[display(fmt = "Remote data fetch has been cancelled")]
+	#[error("Remote data fetch has been cancelled")]
 	RemoteFetchCancelled,
 	/// Remote fetch has been failed.
-	#[display(fmt = "Remote data fetch has been failed")]
+	#[error("Remote data fetch has been failed")]
 	RemoteFetchFailed,
 	/// Error decoding call result.
-	#[display(fmt = "Error decoding call result of {}: {}", _0, _1)]
-	CallResultDecode(&'static str, CodecError),
+	#[error("Error decoding call result of {0}")]
+	CallResultDecode(&'static str, #[source] CodecError),
 	/// Error converting a parameter between runtime and node.
-	#[display(fmt = "Error converting `{}` between runtime and node", _0)]
-	#[from(ignore)]
+	#[error("Error converting `{0}` between runtime and node")]
 	RuntimeParamConversion(String),
 	/// Changes tries are not supported.
-	#[display(fmt = "Changes tries are not supported by the runtime")]
+	#[error("Changes tries are not supported by the runtime")]
 	ChangesTriesNotSupported,
 	/// Error reading changes tries configuration.
-	#[display(fmt = "Error reading changes tries configuration")]
+	#[error("Error reading changes tries configuration")]
 	ErrorReadingChangesTriesConfig,
 	/// Key changes query has failed.
-	#[display(fmt = "Failed to check changes proof: {}", _0)]
-	#[from(ignore)]
+	#[error("Failed to check changes proof: {0}")]
 	ChangesTrieAccessFailed(String),
 	/// Last finalized block not parent of current.
-	#[display(fmt = "Did not finalize blocks in sequential order.")]
-	#[from(ignore)]
+	#[error("Did not finalize blocks in sequential order.")]
 	NonSequentialFinalization(String),
 	/// Safety violation: new best block not descendent of last finalized.
-	#[display(fmt = "Potential long-range attack: block not in finalized chain.")]
+	#[error("Potential long-range attack: block not in finalized chain.")]
 	NotInFinalizedChain,
 	/// Hash that is required for building CHT is missing.
-	#[display(fmt = "Failed to get hash of block for building CHT")]
+	#[error("Failed to get hash of block for building CHT")]
 	MissingHashRequiredForCHT,
 	/// Invalid calculated state root on block import.
-	#[display(fmt = "Calculated state root does not match.")]
+	#[error("Calculated state root does not match.")]
 	InvalidStateRoot,
 	/// Incomplete block import pipeline.
-	#[display(fmt = "Incomplete block import pipeline.")]
+	#[error("Incomplete block import pipeline.")]
 	IncompletePipeline,
-	#[display(fmt = "Transaction pool not ready for block production.")]
+	#[error("Transaction pool not ready for block production.")]
 	TransactionPoolNotReady,
-	#[display(fmt = "Database: {}", _0)]
-	DatabaseError(sp_database::error::DatabaseError),
+	#[error("Database")]
+	DatabaseError(#[from] sp_database::error::DatabaseError),
```

Comment:
Will this be printed?

Reply:
Yes, with `#[from]` it will be printed as the next error in the error chain. `#[from]` implicitly includes `#[source]`